src/Traits/Activatable.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Traits;
+
+trait Activatable
+{
+    /**
+     * Returns true if the user has been activated
+     * and activation is required after registration.
+     */
+    public function isActivated(): bool
+    {
+        // If activation is not required, then we're always active.
+        return ! $this->shouldActivate() || $this->active;
+    }
+
+    /**
+     * Returns true if the user has not been activated.
+     */
+    public function isNotActivated(): bool
+    {
+        return ! $this->isActivated();
+    }
+
+    /**
+     * Activates the user.
+     */
+    public function activate(): bool
```

Comment:
I suggest that it has a parameter, for example public function `activate(...$ids)`.
By default, consider the **current user**, otherwise, if the input `$ids` is set, activate the target users. It is good for the management department and prevents writing a new query.

I have the same opinion for `deactivate()`.

Reply:
`Activatable` is used in `User`. So `activate(...$ids)` is not appropriate.

```php
$user->activate(1, 2, 3)
```
It means a user can activate other users. It is bad design.

If we need the api `activate(...$ids)`, it should be in other class.

Reply:
I agree with @kenjis  that this is not appropriate for this use case.

Reply:
Ok, I realized.